src/GameOfLifeBoard.test.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {cleanup, fireEvent, render} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import GameOfLifeBoard from './GameOfLifeBoard';
+
+afterEach(cleanup);
+
+it('renders without crashing', () => {
+  const boardState = [
+    [1, 0, 0, 1, 0],
+    [0, 1, 1, 1, 1],
+    [0, 0, 0, 1, 0],
+    [1, 0, 1, 1, 0],
+    [1, 0, 0, 0, 1]
+  ];
+  const div = document.createElement('div');
+  ReactDOM.render(<GameOfLifeBoard boardState={boardState}/>, div);
+  ReactDOM.unmountComponentAtNode(div);
+});
+
+it('contains table with board state and buttons for next generation and reset', () => {
+  const boardState = [
+    [1, 0, 0, 1, 0],
+    [0, 1, 1, 1, 1],
+    [0, 0, 0, 1, 0],
+    [1, 0, 1, 1, 0],
+    [1, 0, 0, 0, 1]
+  ];
+  const {baseElement} = render(
+    <GameOfLifeBoard boardState={boardState} />
+  );
+
+  expect(baseElement).toHaveTextContent('1001001111000101011010001Další generace!Resetovat');
+});
+
+it('is able to generate next generation when "Next Generation!" button is clicked', () => {
+  const boardState = [
+    [1, 0, 0, 1, 0],
+    [0, 1, 1, 1, 1],
+    [0, 0, 0, 1, 0],
+    [1, 0, 1, 1, 0],
+    [1, 0, 0, 0, 1]
+  ];
+  const {baseElement, getByText} = render(
+    <GameOfLifeBoard boardState={boardState} />
+  );
+
+  fireEvent.click(getByText('Další generace!'));
+  expect(baseElement).toHaveTextContent('0101101001000000111101010Další generace!Resetovat');
+});
+
+it('is able to reset initial board state when "reset" button is clicked', () => {
+  const boardState = [
+    [1, 0, 0, 1, 0],
+    [0, 1, 1, 1, 1],
+    [0, 0, 0, 1, 0],
+    [1, 0, 1, 1, 0],
+    [1, 0, 0, 0, 1]
+  ];
+  const {baseElement, getByText} = render(
+    <GameOfLifeBoard boardState={boardState} />
+  );
+
+  fireEvent.click(getByText('Další generace!'));
+  expect(baseElement).toHaveTextContent('0101101001000000111101010Další generace!Resetovat');
+
+  fireEvent.click(getByText('Další generace!'));
+  expect(baseElement).toHaveTextContent('0011100111010010101101011Další generace!Resetovat');
+
+  fireEvent.click(getByText('Resetovat'));
+  expect(baseElement).toHaveTextContent('1001001111000101011010001Další generace!Resetovat');
+});
```

Comment:
I think you are testing something else than `GameOfLifeBoard` component here :-). Ideally, `next` function should be mocked if this should be a unit test.

Reply:
That sounds like a good idea. Depends on the "Unit". Would you give me a hint for a mocking tool, @vysinsky ?

Reply:
Jest has it included: https://jestjs.io/docs/en/mock-functions.html

You need to mock the `gol` module and mock implementation of `next` function.

I can help you more (give you the code) if you want, let me know :).

Reply:
And hint: if you change `GameOfLifeBoard` component to receive `next` as prop (so you don't import it in the `GameOfLifeBoard` module mocking gets much easier ;).

Reply:
Thank you for these ideas. :-) Dependency injection is a good way.